### PR TITLE
DAC interface for SAMD51.

### PR DESCRIPTION
**Pull request by BojanSof**
Simple DAC interface for SAMD51 MCU.
Assumes GCLK4 is configured for 12 MHz output .
There is code duplication related to `SUPC` being shared between ADC, DAC and Temperature sensor that needs to be resolved in future PR.

### CODE REVIEW

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+#include <component-version.h> //< Atmel SAM version defines
+#else
+#define COMPONENT_VERSION_MAJOR 1 //< Arduino has v1 headers
+#endif
+
+#include "../Gpio.hpp"
+
+namespace BareCpper
+{
+	namespace SAMD51
```
Comment (CraigHutchinson) [inline include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp]: fix indent

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+     * the reference selection should be UnbufferedExternalReference.
+     * 
+     */
+    enum class DacReference : uint32_t
+    {
+      UnbufferedExternalReference,
+      VoltageSupply,
+      BufferedExternalReference,
+      Internal1V0                 = SUPC_VREF_SEL_1V0_Val,
+			Internal1V1                 = SUPC_VREF_SEL_1V1_Val,
```
Comment (CraigHutchinson) [inline include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp]: fix indent

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+         * Enable the DAC controller and wait for the
+         * startup time of the DACs to pass.
+         * 
+         */
+        void enable()
+        {
+          DAC->CTRLA.bit.ENABLE = 1;
+          while(DAC->SYNCBUSY.bit.ENABLE);
+          // wait for the startup time to finish for both DACs
+          ///@note due to silicon bug, check INTFLAG EMPTY 0 and EMPTY 1
```
Comment (CraigHutchinson) [inline include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp]: Comment: is there a bug reference for the Errata?

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+        void disableChannel()
+        {
+          static_assert(DacIndex < 2, "There are only two dac channels available, 0 and 1");
+          DAC->DACCTRL[DacIndex].reg &= ~DAC_DACCTRL_ENABLE;
+          // config pins
+          // PA02(DAC0) and PA05(DAC1)
+          // disable pmuxen
+          if constexpr(DacIndex == 0)
+          {
+            BareCpper::gpioFunction<PA02>(PA02{}, std::nullopt);
```
Comment (CraigHutchinson) [inline include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp:301]: PA02 is design specific? Can be solved in future if needs more refactor to remove?

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp (hunk 1 above)
Comment (BojanSof) [reply include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp]: Fixed

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp (hunk 2 above)
Comment (BojanSof) [reply include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp]: Fixed

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp (hunk 3 above)
Comment (BojanSof) [reply include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp]: Added comment (errata 2.9.6).

File: include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp (hunk 4 above)
Comment (BojanSof) [reply include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp:301]: The outputs of the DACs are always connected to PA02(DAC0) and PA05(DAC1), there are no other pins on the MCU to be used as DAC outputs.

Comment (CraigHutchinson) [reply include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp:301]: Thanks, that explains this 👍

**Review by CraigHutchinson — APPROVED** [review]

**Pull request merged**